src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -394,6 +389,14 @@ ShouldSyncTableMetadata(Oid relationId)
 	bool citusTableWithNoDistKey =
 		IsCitusTableTypeCacheEntry(tableEntry, CITUS_TABLE_WITH_NO_DIST_KEY);
 
+	/*
+	 * We shouldn't sync metadata for distributed tables created with master_create_distributed_table
+	 */
+	if (hashDistributed && tableEntry->colocationId == INVALID_COLOCATION_ID)
```

Comment:
Otherwise we try to sync metadata for tables created with master_create_distributed_table while adding nodes, but since we have explicit 'viaDeprecatedAPI' check on CreateDistributedTable https://github.com/citusdata/citus/blob/3c834e66936da6b0d76a8e024f21b61783b7ed0d/src/backend/distributed/commands/create_distributed_table.c#L517  we don't have metadata for them. We may add that check as separate PR.

Reply:
We were using pg_dist_partition entries before whil syncing metadata, but we've started to use pg_dist_object with this PR to propagate shell tables.

Reply:
I really don't see a problem for marking these tables as distributed objects as well. In fact, if we don't sync these  tables, I'd expect to get user complains.

Do you see any specific reasons for not doing it?

Reply:
So, basically move:
```C
	if (ShouldSyncTableMetadata(relationId))
	{
		CreateShellTableOnWorkers(relationId);
		MarkObjectDistributed(&tableAddress);
		CreateTableMetadataOnWorkers(relationId);
		CreateInterTableRelationshipOfRelationOnWorkers(relationId);
	}
```

This before we return for `viaDeprecatedAPI` in `CreateDistributedTable`?